util/Mongoose.hx

```diff
@@ -312,21 +325,22 @@ class Mongoose {
 		return objDecl;
 	}
 
-	static function classFieldToSchemaField( f : ClassField ){
+	static function classFieldToSchemaField( f : ClassField, typeKey:String ){
 
 		var type = {
 			pos : f.pos,
-			expr : typeToSchemaType(f.type)
+			expr : typeToSchemaType(f.type, typeKey)
 		};
-
-		var expr = macro { type : $type };
+        
+        if(typeKey.substr(0, 1) == "$")
+            typeKey = "@$__hx__" + typeKey; 
+		var expr = macro { $typeKey: $type };
```

Comment:
Yes, I can see it is somehow correct.
But I'd like to avoid manually adding this weird `@$__hx__` prefix, and leave this to compiler (as it could change in future haxe versions, etc).
If it is not possible to "escape" these field names in any cleaner way, then I guess filing an issue to the haxe compiler would be necessary.
In the meantime, I would recommand removing this hack anyway, and using some easier to handle field such as `__type__`.